components/financial_assistance/app/models/financial_assistance/applicant.rb

```diff
@@ -1466,6 +1466,7 @@ def presence_of_attr_step_1
     def presence_of_attr_other_qns # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity TODO: Remove this
       if is_pregnant
         errors.add(:pregnancy_due_on, "' Pregnancy Due date' should be answered if you are pregnant") if pregnancy_due_on.blank? && FinancialAssistanceRegistry.feature_enabled?(:pregnancy_due_on_required)
+        errors.add(:pregnancy_due_on,"' Pregnancy Due date' should be in the future") if pregnancy_due_on.present? && pregnancy_due_on < Date.today
```

Comment:
Enroll has a TimeKeeper class that is used to track the system date.  We should use it for these kinds of date checks, e.g., `pregnancy_due_on < TimeKeeper.date_of_record`.